test/validation/api/event/event_vector.c

```diff
@@ -0,0 +1,481 @@
+
+	odp_packet_free(pkt);
+	for (uint32_t i = 0; i < num_bufs; i++)
+		odp_buffer_free(buf[i]);
+
+	CU_ASSERT(odp_pool_destroy(pkt_pool) == 0);
+	CU_ASSERT(odp_pool_destroy(buf_pool) == 0);
+}
+
+/*
+ * Test that odp_event_free() and odp_event_vector_free() return the event
+ * vector back to the pool so that it becomes allocatable again.
+ */
+static void test_vector_freeing(void)
+{
+	odp_event_vector_t evv[NUM_VECTORS];
+	odp_event_vector_t extra;
+
+	for (uint32_t i = 0; i < NUM_VECTORS; i++) {
+		evv[i] = odp_event_vector_alloc(evv_pool);
+		CU_ASSERT_FATAL(evv[i] != ODP_EVENT_VECTOR_INVALID);
+	}
+	extra = odp_event_vector_alloc(evv_pool);
+	CU_ASSERT(extra == ODP_EVENT_VECTOR_INVALID);
+
+	odp_event_vector_free(evv[0]);
+	evv[0] = odp_event_vector_alloc(evv_pool);
+	CU_ASSERT_FATAL(evv[0] != ODP_EVENT_VECTOR_INVALID);
+	extra = odp_event_vector_alloc(evv_pool);
+	CU_ASSERT(extra == ODP_EVENT_VECTOR_INVALID);
+
+	odp_event_free(odp_event_vector_to_event(evv[0]));
+	evv[0] = odp_event_vector_alloc(evv_pool);
+	CU_ASSERT_FATAL(evv[0] != ODP_EVENT_VECTOR_INVALID);
+	extra = odp_event_vector_alloc(evv_pool);
+	CU_ASSERT(extra == ODP_EVENT_VECTOR_INVALID);
+
+	for (uint32_t i = 0; i < NUM_VECTORS; i++)
+		odp_event_vector_free(evv[i]);
+}
+
+/*
+ * Test that two event vectors allocated from the same pool have
+ * different handles, user areas (if supported), user flags, sizes
+ * and event tables.
+ */
+static void test_uniqueness(void)
+{
+	odp_event_vector_t evv1, evv2;
+	odp_event_t *ev_tbl1, *ev_tbl2;
+
+	ODP_STATIC_ASSERT(NUM_VECTORS >= 2, "vector pool is too small");
+	evv1 = odp_event_vector_alloc(evv_pool);
+	evv2 = odp_event_vector_alloc(evv_pool);
+	CU_ASSERT_FATAL(evv1 != ODP_EVENT_VECTOR_INVALID);
+	CU_ASSERT_FATAL(evv2 != ODP_EVENT_VECTOR_INVALID);
+
+	CU_ASSERT(evv1 != evv2);
+	CU_ASSERT(uarea_size == 0 || (odp_event_vector_user_area(evv1) !=
+				      odp_event_vector_user_area(evv2)));
+	CU_ASSERT(odp_event_vector_to_u64(evv1) !=
+		  odp_event_vector_to_u64(evv2));
+
+	CU_ASSERT(odp_event_vector_tbl(evv1, &ev_tbl1) == 0);
+	CU_ASSERT(odp_event_vector_tbl(evv2, &ev_tbl2) == 0);
+	CU_ASSERT(ev_tbl1 != ev_tbl2);
+
+	CU_ASSERT(odp_event_vector_user_flag(evv1) == 0);
+	CU_ASSERT(odp_event_vector_user_flag(evv2) == 0);
+	odp_event_vector_user_flag_set(evv1, 1);
+	CU_ASSERT(odp_event_vector_user_flag(evv1) == 1);
+	CU_ASSERT(odp_event_vector_user_flag(evv2) == 0);
+
+	CU_ASSERT(odp_event_vector_size(evv1) == 0);
+	CU_ASSERT(odp_event_vector_size(evv2) == 0);
+	odp_event_vector_size_set(evv1, 1);
+	CU_ASSERT(odp_event_vector_size(evv1) == 1);
+	CU_ASSERT(odp_event_vector_size(evv2) == 0);
+	odp_event_vector_size_set(evv1, 0);
+
+	odp_event_vector_free(evv1);
+	odp_event_vector_free(evv2);
+}
+
+int evv_suite_init(void)
+{
+	odp_pool_capability_t capa;
+	static odp_pool_param_t param; /* keep around until tests finished */
+
+	if (odp_pool_capability(&capa) != 0) {
+		printf("\nodp_pool_capability() failed\n");
```

Comment:
Generally the validation tests use `ODPH_ERR()` macro.

Reply:
changed in v12